symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java

```diff
@@ -57,14 +74,10 @@ public void start() throws AuthUnauthorizedException, ApiException {
                 this.started.set(true);
                 this.readDatafeed();
             } while (this.started.get());
-        } catch (Throwable e) {
-            if (e instanceof ApiException) {
-                throw (ApiException) e;
-            } else if (e instanceof AuthUnauthorizedException) {
-                throw (AuthUnauthorizedException) e;
-            } else {
-                log.error("Unknown error", e);
-            }
+        } catch (AuthUnauthorizedException | ApiException exception) {
+            throw exception;
+        } catch (Throwable throwable) {
+            log.error("Unknown error", throwable);
```

Comment:
Do we know what kind of errors it could be ? Do we have a reason why we want to silently ignore these errors ?

Reply:
AuthUnauthorizedException and ApiException exception are the only two exception can happen in our case. In spite of it, the `Retry#executeCheckedSupplier` still throws a Throwable because it didn't care about what we did inside the retried block. That's why we have this catch here, although in fact, it is never caught.